activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -90,6 +90,48 @@ def sql_color(sql)
       def logger
         ActiveRecord::Base.logger
       end
+
+      def debug(progname = nil, &block)
+        return unless super
+
+        if ActiveRecord::Base.verbose_query_logs
+          log_query_source
+        end
+      end
+
+      def log_query_source
+        source_line, line_number = extract_callstack(caller_locations)
+
+        if source_line
+          if defined?(::Rails) && defined?(::Rails.root)
```

Comment:
You don't need two checks here - `defined?` can handle the whole expression at once:

```
$ irb
irb(main):001:0> defined?(::Rails.root)
=> nil
```

Reply:
Makes sense, updated.